### PR TITLE
tvb-contrib changes by DP

### PR DESCRIPTION
**Pull request by dionperd**
Pull request to merge to master. 

Most changes are expected to relate to tvb.contrib.

### CODE REVIEW

File: tvb_contrib/tvb/contrib/scripts/plot/power_spectra_interactive.py [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+
+# Define a colour theme... see: matplotlib.colors.cnames.keys()
+BACKGROUNDCOLOUR = "slategrey"
+EDGECOLOUR = "darkslateblue"
+AXCOLOUR = "steelblue"
+BUTTONCOLOUR = "steelblue"
+HOVERCOLOUR = "blue"
+
+
+class PowerSpectraInteractive(HasTraits):
```
Comment (liadomide) [inline tvb_contrib/tvb/contrib/scripts/plot/power_spectra_interactive.py:78]: Why do you need to copy this class in contrib?
Can't you at least inherit from the similarly named class from under `tvb.simulator.plot`?

Comment (dionperd) [reply tvb_contrib/tvb/contrib/scripts/plot/power_spectra_interactive.py:78]: Let me try! I remember that there were smaller or bigger changes all over that class. I also found some bugs there. But, I need to look at it again more carefully. I will come back to you about this.

Comment (liadomide) [reply tvb_contrib/tvb/contrib/scripts/plot/power_spectra_interactive.py:78]: we would happily accept bug fixes in `tvb.simulator.plot` itself.
I think this widget was build mostly for your usages, thus no harm to change it as you like, as long as it still works.

Comment (dionperd) [reply tvb_contrib/tvb/contrib/scripts/plot/power_spectra_interactive.py]: > we would happily accept bug fixes in `tvb.simulator.plot` itself. I think this widget was build mostly for your usages, thus no harm to change it as you like, as long as it still works.

So, I have inherited it from tvb.simulator.plot and also renamed it as my widget plots also coherence.

Comment (liadomide) [reply tvb_contrib/tvb/contrib/scripts/plot/power_spectra_interactive.py]: Cool about the inherit. I still see some place to improve, for example: can't the `PowerSpectraCoherenceInteractive.__init__` and `PowerSpectraCoherenceInteractive.create_figure` methods be smaller (by removing the duplicate code which already exists in the parent)?
Also, sonar complains on 14 code smells. Some are quite easy to fix before we merge (e.g. remove commented code)

Comment (dionperd) [reply tvb_contrib/tvb/contrib/scripts/plot/power_spectra_interactive.py]: > Cool about the inherit. I still see some place to improve, for example: can't the `PowerSpectraCoherenceInteractive.__init__` and `PowerSpectraCoherenceInteractive.create_figure` methods be smaller (by removing the duplicate code which already exists in the parent)? Also, sonar complains on 14 code smells. Some are quite easy to fix before we merge (e.g. remove commented code)

Yes, I could reduce the duplicate code, but only if I refactor the original tvb.simulator.plot class.

So, let me try. I will also, then, check the smelly code...!

**Review by liadomide — APPROVED** [review]

**Comment by liadomide** [comment]
Looks decent to me. @dionperd should we proceed and merge or do you still have changes planned for these parts?

**Review by liadomide — APPROVED** [review]

**Review by romina1601 — APPROVED** [review]

**Review by liadomide — APPROVED** [review]
I did a rebase of the branch commits against master to have a cleaner history.
We should merge.
Another PR will come with adding unit-tests in tvb-contrib

**Pull request merged**